### PR TITLE
Add middleware to check permissions of logged in user

### PR DESCRIPTION
**Pull request by SquarePlayn**
## Proposed changes

This PR adds a middleware class `CheckPermission`, usable as middleware `permission:{permission}`, that can be used to protect a route to ensure that the user is logged in and has a certain Auth0 permission.

Fixes [AB#28](https://dev.azure.com/jdokter/973dd796-5b61-4e4a-a52b-ab66f88cef8a/_workitems/edit/28).

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Unit testing
As usual

### Manual testing
Use the package in a laravel app and protect an endpoint with the permissions middleware.

## Further comments
In order for an access token to be attached to the logged in session, the access tokens have to be persisted. This is a setting in the Auth0 child package's config, which by default is off/false. Thus, this package also publishes this config, with the value set to true. The package is also robust against if you forget this, for example for existing projects that already have their config published where it is forgotten to change when this new middleware is used.

### CODE REVIEW

File: src/Http/Middleware/CheckPermissions.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public function handle(Request $request, Closure $next, string $permissionRequired = null)
+    {
+        // Make sure the user is logged in
+        if (!Auth::check()) {
+            return redirect()->route('login');
+        }
+
+        // Fetch the access token of the user
+        $token = Auth::user()->getAuthPassword();
```
Comment (marijnvanderhorst) [inline src/Http/Middleware/CheckPermissions.php:33]: To me it doesn't really make sense to use the access token for this purpose. The access token should be used to access resources on behalf of the user. What we are doing here is checking the permissions of the logged-in user, which should be performed using the ID token instead, right? (following the OpenID connect protocol)

**Review by marijnvanderhorst — CHANGES_REQUESTED** [review]
Please update the readme as well with a short explanation on how to use this new middleware.

**Review by marijnvanderhorst — APPROVED** [review]
Accepting my own changes.. :)

**Comment by SquarePlayn** [comment]
> @SquarePlayn I've updated it to use the ID token (/ userinfo) now instead of the Bearer token. Please check whether you agree with my changes.

I have made some small changes to the tests, which contained something with the JWT token that is not needed anymore and made one test a little stronger of a test. Other than that, looks good, thanks for the help!

**Review by marijnvanderhorst — APPROVED** [review]

**Pull request merged**